elixir/lib/app_identity/plug.ex

```diff
@@ -275,19 +291,37 @@ if Code.ensure_loaded?(Plug.Conn) do
         headers: get_headers(options),
```

Comment:
(Opinion/future) Should the option processing be its own module, this one is getting big, and the types & logic of checking the options are a bit different from what the plug's doing.

I'd be tempted to do a refactor as a separate piece of work to see what it looks like.